Orange/widgets/data/owfile.py

```diff
@@ -1,16 +1,23 @@
 import os
+from itertools import chain
+from itertools import count
 from warnings import catch_warnings
+import numpy as np
 from xlrd import open_workbook
 from PyQt4 import QtGui, QtCore
 from PyQt4.QtGui import QSizePolicy as Policy
 
 from Orange.canvas.gui.utils import OSX_NSURL_toLocalFile
+from Orange.data import Domain
 from Orange.widgets import widget, gui
-from Orange.widgets.settings import Setting, ContextHandler, ContextSetting
+from Orange.widgets.settings import Setting, ContextHandler, ContextSetting, \
+    PerfectDomainContextHandler
+from Orange.widgets.utils.domaineditor import DomainEditor
 from Orange.widgets.utils.itemmodels import PyListModel
 from Orange.widgets.utils.filedialogs import RecentPathsWComboMixin
+from Orange.data import DiscreteVariable, StringVariable
```

Comment:
This could also be merged with the import of Domain (from the same module) above.

It is not so noticeable, because the imports are not sorted - could you do that as well, while you are changing things here (sort everything inside the 3 groups of imports).